txcache/wrappedTransaction.go

```diff
@@ -69,3 +72,7 @@ func (wrappedTx *WrappedTransaction) isTransactionMoreValuableForNetwork(otherTr
 	// In the end, compare by transaction hash
 	return bytes.Compare(wrappedTx.TxHash, otherTransaction.TxHash) < 0
 }
+
+func (wrappedTx *WrappedTransaction) isFeePayerSameAsSender() bool {
+	return bytes.Equal(wrappedTx.FeePayer, wrappedTx.Tx.GetSndAddr())
```

Comment:
Maybe `len(asRelayed.GetRelayerAddr())` (see above) can be checked to skip the comparison on the most common flow. But then again, perhaps this is a micro-optimization, sorry. Can also be kept in mind for a future PR.